addons/stock/models/stock_quant.py

```diff
@@ -45,13 +77,15 @@ class StockQuant(models.Model):
     quantity = fields.Float(
         'Quantity',
         help='Quantity of products in this quant, in the default unit of measure of the product',
-        readonly=True, required=True, oldname='qty')
+        readonly=True, oldname='qty')
+    inventory_quantity = fields.Float('Inventoried Quantity')
```

Comment:
i don't remember why this field is stored?

Reply:
write 3 tests: create a quant and set a qty
edit a quant and lessen the qty
edit a quant and increase the qty
then make this field compute, male an inverse method creating the stock moves, remove the override of create and write